x-pack/plugins/watcher/public/sections/watch_edit/components/threshold_watch_edit/threshold_watch_edit.tsx

```diff
@@ -262,8 +279,16 @@ const ThresholdWatchEditUi = ({ intl, pageTitle }: { intl: InjectedIntl; pageTit
                 onChange={async (selected: EuiComboBoxOptionProps[]) => {
                   setWatchProperty('index', selected.map(aSelected => aSelected.value));
                   const indices = selected.map(s => s.value as string);
+
+                  // reset the time field if indices are deleted
+                  if (indices.length === 0) {
+                    setTimeFieldOptions(getTimeFieldOptions([]));
+                    setWatchProperty('timeFields', []);
+                    return;
+                  }
                   const theFields = await getFields(indices);
                   setFields(theFields);
+                  setWatchProperty('timeFields', theFields);
```

Comment:
Minor nit: can we use a name that identifies these fields a bit more than "theFields"? e.g. "timeFields"?

Reply:
Agreed, it is a little confusing. This is used in a couple different places of the code in reference to all ES fields, and then there is some manipulation in areas to filter only time fields. I renamed this to `esFields`, and `timeFields`, where used. Let me know if you think that's clearer.